include/bluetooth/gap.h

```diff
@@ -189,6 +190,139 @@ enum {
 	BT_GAP_SCA_0_20 = 7,
 };
 
+/**
+ * @brief Encode 64-bit LE Supported Features into array values in little-
+ *        endian format.
+ *
+ * Helper macro to encode 64-bit LE Supported Features value into advertising
+ * data.
+ *
+ * Example of how to encode the `0xF00DF00DF00DF00D` into advertising data.
+ *
+ * @code
+ * BT_DATA_BYTES(BT_DATA_LE_SUPPORTED_FEATURES, BT_LE_SUPP_FEAT_64_ENCODE(0xF00DF00DF00DF00D))
+ * @endcode
+ *
+ * @param w64 LE Supported Features value (64-bits)
+ *
+ * @return The comma separated values for LE Supported Features value that
+ *         may be used directly as an argument for @ref BT_DATA_BYTES.
+ */
+#define BT_LE_SUPP_FEAT_64_ENCODE(w64) \
+	(((w64) >> 0) & 0xFF),  \
+	(((w64) >> 8) & 0xFF),  \
+	(((w64) >> 16) & 0xFF), \
+	(((w64) >> 24) & 0xFF), \
+	(((w64) >> 32) & 0xFF), \
+	(((w64) >> 40) & 0xFF), \
+	(((w64) >> 48) & 0xFF), \
+	(((w64) >> 56) & 0xFF)
+
+/** @brief Encode 4 least significant bytes of 64-bit LE Supported Features into
+ *         4 bytes long array of values in little-endian format.
+ *
+ *  Helper macro to encode 64-bit LE Supported Features value into advertising
+ *  data. The macro encodes 4 least significant bytes into advertising data.
+ *  Other 4 bytes are not encoded.
+ *
+ *  Example of how to encode the `0xF00DF00DF00DF00D` into advertising data.
+ *
+ *  @code
+ *  BT_DATA_BYTES(BT_DATA_LE_SUPPORTED_FEATURES, BT_LE_SUPP_FEAT_64_ENCODE(0xF00DF00DF00DF00D))
+ *  @endcode
+ *
+ * @param w64 LE Supported Features value (64-bits)
+ *
+ * @return The comma separated values for LE Supported Features value that
+ *         may be used directly as an argument for @ref BT_DATA_BYTES.
+ */
+#define BT_LE_SUPP_FEAT_32_ENCODE(w64) \
+	(((w64) >> 0) & 0xFF),  \
+	(((w64) >> 8) & 0xFF),  \
+	(((w64) >> 16) & 0xFF), \
+	(((w64) >> 24) & 0xFF)
```

Comment:
Given that this is very similar to `BT_UUID_32_ENCODE`, we should probably (in another PR) actually just create some generic macros to encode values into arrays (similar to `sys_put_be16`/`sys_put_le16`).

Reply:
> I'm not sure what the use case is for having macros for 8, 4, 3, 2 and 1 bytes. Can you elaborate? If we have e.g. 3, why not 5?

According to the BT 5.3. Core specification, the features are stored in 64bit `uint`. As of now, 40 bits have assigned values, other are RFU.  To store features that are in bits 32-39 one can use `BT_LE_SUPP_FEAT_64_ENCODE`. In this situation the most suitable type is `uint64_t`. For bits 24-31 there is a macro `BT_LE_SUPP_FEAT_32_ENCODE`; and so one...

This makes it easier to encode the features value into adv. data following rules that: 
- the size is changeable 0 to more octets long
- all 0x0 octets after last non zero octet shall be omitted from the value transmitted.

> Given that this is very similar to BT_UUID_32_ENCODE, we should probably (in another PR) actually just create some generic macros to encode values into arrays (similar to sys_put_be16/sys_put_le16).

Agree to that.

Reply:
> As of now, 40 bits have assigned values, other are RFU.

Shouldn't we have a `BT_LE_SUPP_FEAT_40_ENCODE` encode then? 

> all 0x0 octets after last non zero octet shall be omitted from the value transmitted.

This makes it sound like the `BT_LE_SUPP_FEAT_64_ENCODE` will break that rule

Reply:
Yes you are right. I have to change `BT_LE_SUPP_FEAT_64_ENCODE` into `BT_LE_SUPP_FEAT_40_ENCODE`.

Reply:
Change applied.